packages/dom/src/document.rs

```diff
@@ -243,6 +265,71 @@ impl Document {
             .or(self.try_root_element().map(|el| el.id))
     }
 
+    /// Find the label's bound input elements:
+    /// the element id referenced by the "for" attribute of a given label element
+    /// or the first input element which is nested in the label
+    /// Note that although there should only be one bound element,
+    /// we return all possibilities instead of just the first
+    /// in order to allow the caller to decide which one is correct
+    pub fn label_bound_input_elements(&self, label_node_id: usize) -> Vec<&Node> {
+        let label_node = self.get_node(label_node_id).unwrap();
+        let label_element = label_node.element_data().unwrap();
+        if let Some(target_element_dom_id) = label_element.attr(local_name!("for")) {
+            self.tree()
+                .into_iter()
+                .filter_map(|(_id, node)| {
+                    let element_data = node.element_data()?;
+                    if element_data.name.local != local_name!("input") {
+                        return None;
+                    }
+                    let id = element_data.id.as_ref()?;
+                    if *id == *target_element_dom_id {
+                        Some(node)
+                    } else {
+                        None
+                    }
+                })
+                .collect()
```

Comment:
We probably ought to maintain a global ID map so this can be looked up much more quickly? I wouldn't block this PR on that, but perhaps an "iterate over nodes with a given id" method could be added to `Document`?